specs/core/0_fork-choice.md

```diff
@@ -130,13 +144,43 @@ def get_head(store: Store) -> Hash:
         head = max(children, key=lambda root: (get_latest_attesting_balance(store, root), root))
 ```
 
+#### `should_update_justified_checkpoint`
+
+```python
+def should_update_justified_checkpoint(store: Store, justified_checkpoint: Checkpoint) -> bool:
+    if get_current_slot(store) % SLOTS_PER_EPOCH < SAFE_SLOTS_TO_UPDATE_JUSTIFIED:
+        return True
+
+    justified_block = store.blocks[justified_checkpoint.root]
+    if justified_block.slot <= compute_start_slot_at_epoch(store.justified_checkpoint.epoch):
```

Comment:
Quite confusing here with `justified_checkpoint` vs `store.justified_checkpoint`. Was thinking "can this even hit this branch" before I noticed. Maybe change the argument name?